src/Dropdown.php

```diff
@@ -4,359 +4,696 @@
+use Yiisoft\Html\Tag\Button;
+use Yiisoft\Html\Tag\Div;
+use Yiisoft\Html\Tag\Span;
 use Yiisoft\Html\Tag\Ul;
 
-use function array_key_exists;
-use function array_merge;
-use function array_merge_recursive;
-use function array_unique;
-use function is_array;
-use function is_string;
-
 /**
  * Dropdown renders a Bootstrap dropdown menu component.
  *
  * For example,
  *
  * ```php
- * <div class="dropdown">
- *     <?php
- *         echo Dropdown::widget()
- *             ->items([
- *                 ['label' => 'DropdownA', 'url' => '/'],
- *                 ['label' => 'DropdownB', 'url' => '#'],
- *             ]);
- *     ?>
- * </div>
+ * echo Dropdown::widget()
+ *     ->items(
+ *         DropdownItem::link('Action', '#'),
+ *         DropdownItem::link('Another action', '#'),
+ *         DropdownItem::link('Something else here', '#'),
+ *         DropdownItem::divider(),
+ *         DropdownItem::link('Separated link', '#'),
+ *     )
+ *     ->toggleContent('Toggle dropdown')
+ *     ->toggleVariant(DropdownToggleVariant::DANGER)
+ *     ->toggleSplit()
+ *     ->toggleSplitContent('Danger')
+ *     ->toggleSizeLarge()
+ *     ->render();
  * ```
  */
-final class Dropdown extends Widget
+final class Dropdown extends \Yiisoft\Widget\Widget
 {
-    public const ALIGNMENT_END = 'dropdown-menu-end';
-    public const ALIGNMENT_SM_END = 'dropdown-menu-sm-end';
-    public const ALIGNMENT_MD_END = 'dropdown-menu-md-end';
-    public const ALIGNMENT_LG_END = 'dropdown-menu-lg-end';
-    public const ALIGNMENT_XL_END = 'dropdown-menu-xl-end';
-    public const ALIGNMENT_XXL_END = 'dropdown-menu-xxl-end';
-    public const ALIGNMENT_SM_START = 'dropdown-menu-sm-start';
-    public const ALIGNMENT_MD_START = 'dropdown-menu-md-start';
-    public const ALIGNMENT_LG_START = 'dropdown-menu-lg-start';
-    public const ALIGNMENT_XL_START = 'dropdown-menu-xl-start';
-    public const ALIGNMENT_XXL_START = 'dropdown-menu-xxl-start';
-
-    private array|string|Stringable|null $items = [];
-    private bool $encodeLabels = true;
-    private bool $encodeTags = false;
-    private array $submenuOptions = [];
-    private array $options = [];
-    private array $itemOptions = [];
-    private array $linkOptions = [];
-    private array $alignment = [];
-
-    private ?string $tagName = null;
-
-    public function getId(?string $suffix = '-dropdown'): ?string
+    private const DROPDOWN_CLASS = 'dropdown';
+    private const DROPDOWN_LIST_CLASS = 'dropdown-menu';
+    private const DROPDOWN_TOGGLE_BUTTON_CLASS = 'btn';
+    private const DROPDOWN_TOGGLE_CLASS = 'dropdown-toggle';
+    private const DROPDOWN_TOGGLE_CONTAINER_CLASS = 'btn-group';
+    private const DROPDOWN_TOGGLE_SPAN_CLASS = 'visually-hidden';
+    private const DROPDOWN_TOGGLE_SPLIT_CLASS = 'dropdown-toggle-split';
+    private const NAME = 'dropdown';
+    private array $alignmentClasses = [];
+    private array $attributes = [];
+    private array $cssClass = [];
```

Comment:
Which suggests, Drodown is a complex class, since it is used in multiple menus, being able to manage the classes separately allows for more control over HTML generation.

Reply:
Correct. Here my comment is just about naming: plural vs singular.

Reply:
done.